static/js/contextMenu.js

```diff
@@ -0,0 +1,197 @@
+      selectedLineNumber = null;
+      contextMenu.hide();
+    }
+  }
+  
+  contextMenu.aceKeyEvent = function(hook, context){
+    if((context.evt.key == "Enter"||context.evt.key == "Backspace") && context.rep.alines.length != lines){
+      lines = context.rep.alines.length;
+      attachContextMenu();
+    }
+  }
+  
+  function onRightClick(lineNumber) {
+    if (selectedLineNumber) {
+      return;
+    }
+    var innerDocBody = $('iframe[name="ace_outer"]').contents().find('iframe').contents().find("#innerdocbody");
+    innerDocBody.children().eq(lineNumber).selectText();
+    selectedLineNumber = lineNumber;
+  }
+  
+  
+  function drawNotifyModal() {
+    var padOuter = $('iframe[name="ace_outer"]').contents().find("body");
+    if (padOuter.find('#notifyModal').length === 0) {
+      var baseURL = window.location.href.slice(0, window.location.href.split('/', 3).join('/').length);
+      var onNotify = function(loginOrEmails) {
+        var copiedText = getTextToCopy();
+        $.ajax({
+          method: 'POST',
+          url: baseURL +'/mypads/api/notify-users',
+          data: {
+            auth_token: localStorage.getItem('token'),
+            url: copiedText.url,
+            text: copiedText.text,
+            loginsOrEmails: loginOrEmails
+          },
+          success: function(data, textStatus, jqXHR) {
+            if (data.success === true) {
+              console.log('notifying success!');
+            } else {
+              console.log('notifying failed');
+            }
+          }
+        });
+      }
+      notifyModal.init(onNotify);
+    }
+    notifyModal.toggle();
+  }
+  
+  function getTextToCopy() {
+    var innerDocWindow = $('iframe[name="ace_outer"]').contents().find('iframe')[0].contentWindow;
+    return {
+      url: window.location.href.slice(0, window.location.href.indexOf('?')) + '?lineNumber=' + (selectedLineNumber + 1),
+      text: innerDocWindow.getSelection().toString()
+    }
+  }
+  
+  function drawContextMenu(x, y){
+    var padOuter = $('iframe[name="ace_outer"]').contents().find("body");
+    var contextMenu = padOuter.find("#context_menu");
+    var newX = (x + 60) + "px";
+    var newY = (y + 20) + "px";
+    var items = [
+      {
+        label: "Copy Link",
+        onclick: function() {
+          var padOuter = $('iframe[name="ace_outer"]').contents().find("body");
+          var $textarea = $("<textarea>", {id: "text_to_copy"});
+          var copiedText = getTextToCopy();
+          $textarea.val(copiedText.text + '\n' + copiedText.url);
+          padOuter.append($textarea);
+          $textarea.copyText();
+          $textarea.remove();
+        }
+      },
+      {
+        label: "Email selected text",
+        onclick: function() {
+          drawNotifyModal();
+        }
+      }
+    ]
+    if(contextMenu.length === 0){
+      padOuter.append("<div id ='context_menu'></div>");
+      contextMenu = padOuter.find("#context_menu");
```

Comment:
Cache the result of $(contextMenu) into a variable.  The $() operation is a costly one that wraps an html element with the necessary convenience methods.